castxml: unbreak binary by adding llvm to buildInputs

bcdarwin opened the pull request:
Adds LLVM to `buildInputs`.

Also sets `CLANG_RESOURCE_DIR` correctly and enables tests.

Fixes #318055.

## Description of changes

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

bcdarwin committed:
castxml: unbreak binary by adding llvm to buildInputs

Fixes #318055.

paparodeo reviewed (approved):
LGTM

could add testers to the argument set and
```nix
  passthru.tests = testers.testVersion {
    package = finalAttrs.finalPackage;
  };
```
to prevent from happening again.

bcdarwin committed:
castxml: use `testers.testVersion` in `passthru.tests`

(in order to prevent CLI regressions such as https://github.com/NixOS/nixpkgs/pull/318055)

paparodeo commented on pkgs/by-name/ca/castxml/package.nix:
```diff
@@ -66,6 +66,10 @@ stdenv.mkDerivation (finalAttrs: {
     runHook postCheck
   '';
 
+  passthru.tests = testers.testVersion {
```
need to add `testers` as an input at the top of the file.

bcdarwin replied:
Yep, added.

paparodeo reviewed (approved):
LGTM: `nix-build -A castxml.tests` succeeds on x64 linux/darwin

AndersonTorres commented on pkgs/by-name/ca/castxml/package.nix:
```diff
@@ -40,6 +40,7 @@ stdenv.mkDerivation (finalAttrs: {
     libffi
     libxml2
     zlib
+    llvm
```
```suggestion
    llvm
    zlib
```

Alphabetical order, since order is relevant in lists.

bcdarwin replied:
Done.

bcdarwin committed:
castxml: fix CLANG_RESOURCE_DIR; enable tests

bcdarwin commented:
Added another commit to unbreak the test suite.

AndersonTorres commented on pkgs/by-name/ca/castxml/package.nix:
```diff
@@ -39,32 +39,28 @@ stdenv.mkDerivation (finalAttrs: {
 
-  cmakeFlags = [
-    (lib.cmakeOptionType "path" "CLANG_RESOURCE_DIR" "${lib.getDev libclang}")
+  cmakeFlags = let clangVersionMajor = builtins.elemAt (lib.strings.splitString "." libclang.version) 0; in [
```
```suggestion
  cmakeFlags = let
    clangVersionMajor = builtins.elemAt (lib.strings.splitString "." libclang.version) 0; 
  in [
```

P.S.: isn't better to use `lib.major`?

https://ryantm.github.io/nixpkgs/functions/library/versions/#function-library-lib.versions.major

AndersonTorres replied:
P.P.S.: there is few to no need of using blocks of indirectio,  when the function is already on `lib` scope.

bcdarwin replied:
Done.

bcdarwin commented:
Result of `nixpkgs-review pr 318140` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>castxml</li>
    <li>gnss-sdr</li>
    <li>gnuradio</li>
    <li>gnuradioPackages.osmosdr</li>
    <li>gnuradioPackages.osmosdr.dev</li>
    <li>ns-3</li>
    <li>python311Packages.pygccxml</li>
    <li>python311Packages.pygccxml.dist</li>
    <li>python312Packages.pygccxml</li>
    <li>python312Packages.pygccxml.dist</li>
  </ul>
</details>
